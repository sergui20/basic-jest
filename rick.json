{
    "id": 1,
    "name": "Rick Sanchez",
    "status": "Death",
    "species": "Human",
    "gender": "Male"
}