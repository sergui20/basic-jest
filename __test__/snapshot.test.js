import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Test with snapshots', () => {
    test('Rick Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    test('Will always fail snapshot', () => {
        const user = {
            createdAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }

        expect(user).toMatchSnapshot();
    });

    test('Allow exceptions', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Oscar Vargas"
        }

        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        });
    });
});
