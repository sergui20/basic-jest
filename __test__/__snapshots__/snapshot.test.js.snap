// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test with snapshots Allow exceptions 1`] = `
Object {
  "id": Any<Number>,
  "name": "Oscar Vargas",
}
`;

exports[`Test with snapshots Rick Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Death",
}
`;

exports[`Test with snapshots Will always fail snapshot 1`] = `
Object {
  "createdAt": 2019-10-10T23:13:58.338Z,
  "id": 1,
}
`;
